common/src/main/java/ru/coderedwolf/wordlearn/common/ui/BaseFragment.kt

```diff
@@ -51,10 +57,25 @@ abstract class BaseFragment : MvpAppCompatFragment() {
         else -> isRealRemoving()
     }
 
+    private val featureLifecycleScopeProvider = FeatureLifecycleScopeProvider()
+
+    private val featureDisposeCompositeDisposable = CompositeDisposable()
+
     internal fun isRealRemoving(): Boolean = (isRemoving && !instanceStateSaved) || (parentFragment as BaseFragment).isRealRemoving()
 
     @CallSuper
-    open fun onRealRemoving() = clearInjector(fragmentComponentName)
+    open fun onRealRemoving(){
+        featureLifecycleScopeProvider.onDestroy()
+        featureDisposeCompositeDisposable.clear()
```

Comment:
we need to use dispose instead clear, cause clear disposes disposables in container and we can still add some disposable to the compositeDisposable